core/src/main/java/org/apache/druid/timeline/partition/SingleElementPartitionChunk.java

```diff
@@ -93,7 +93,7 @@ public boolean equals(Object o)
   @Override
   public int hashCode()
   {
-    return element != null ? element.hashCode() : 0;
+    return 0;
```

Comment:
What's the reasoning behind the hashCode change?

Reply:
This was a bug because of the wacky way `PartitionChunk` is compared. It compares only the partitionId without checking the actual `element`. See `equals()`.